internal/common/conversion/type_conversion.go

```diff
@@ -56,3 +56,10 @@ func IsStringPresent(strPtr *string) bool {
 func MongoDBRegionToAWSRegion(region string) string {
 	return strings.ReplaceAll(strings.ToLower(region), "_", "-")
 }
+
+func NonEmptySliceToSlicePtr[T any](v []T) *[]T {
```

Comment:
if slices only let's be more correct about it 
```suggestion
func NonEmptySliceToSlicePtr[S ~[]E, E any]](v []S) *S {
```

Reply:
name changed to NonEmptyToPtr as it might called for slices or also arrays.